### ISSUE TITLE
Add project level secrets

### ISSUE THREAD

**Issue by Frassle** [pulumi/pulumi#11549]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Add secrets at the project level, this will be fully supported by the Pulumi Service in the same way as stack config (you'll also be able to use any other secret manager).

**Comment by LouisAmon** [comment]
Maybe we can consider this resolved now that Pulumi ESC is released ?

https://www.pulumi.com/docs/pulumi-cloud/esc/get-started/

**Comment by gunzy83** [comment]
> Maybe we can consider this resolved now that Pulumi ESC is released ?
> 
> https://www.pulumi.com/docs/pulumi-cloud/esc/get-started/

Only if you use the Pulumi service. Not sure how many users do not use the Pulumi service and would appreciate all stack config features working at the project level.

**Comment by mattfysh** [comment]
For anyone looking at this feature, the best thing to do is avoid using project-level config as Pulumi seem to have abandoned it, for example - they still have not added a CLI command to `pulumi config set` into the project and not the stack

To batch apply config, try this script

```sh
stacks=$(pulumi stack ls | tail -n +2 | awk '{print $1}' | sed 's/\*$//')
for stack in $stacks; do
  pulumi config set --stack $stack provider:apikey $value --secret
done
```

**Comment by gunzy83** [comment]
> For anyone looking at this feature, the best thing to do is avoid using project-level config as Pulumi seem to have abandoned it, for example - they still have not added a CLI command to `pulumi config set` into the project and not the stack

I respectfully disagree. While they have not added more features (like this one), that does not mean that it is abandoned. Until I see a deprecation notice on [this tiny section of the project file docs](https://www.pulumi.com/docs/iac/concepts/projects/project-file/#config-options), project config still has its place for sensible defaults so that you do not have to specify every config option on every stack. Note: to my knowledge, none of the options in that file have a CLI get/set command.

**Comment by sergioaten** [comment]
seems deprecated or what reference this?
https://www.pulumi.com/docs/iac/concepts/projects/project-file/#deprecated-attributes

**Comment by Frassle** [comment]
> seems deprecated or what reference this?

That was an old attribute where you could use "config" to set where config files where kept relative to the Pulumi.yaml. No one really used it so we repurposed the "config" key for project level config instead.

**Comment by gunzy83** [comment]
> seems deprecated or what reference this? https://www.pulumi.com/docs/iac/concepts/projects/project-file/#deprecated-attributes

My link is to what currently works. I remember when the config option switched not long after I started using Pulumi, and it has been a godsend for setting sensible defaults.
